src/k8s-engine/lima.ts

```diff
@@ -1840,20 +1846,24 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
       const fileContents = `CREDFWD_AUTH=${ leadingDollarSign }'${ stateInfo.user }:${ escapedPassword }'
 CREDFWD_URL='http://${ hostIPAddr }:${ stateInfo.port }'
 `;
-      const credfwdDir = '/etc/rancher/desktop';
-      const credfwdFile = `${ credfwdDir }/credfwd`;
-      const configContents = `{
-  "credsStore": "rancher-desktop"
-}
-`;
+      const defaultConfig = { credsStore: 'rancher-desktop' };
+      let configContents: string;
 
-      await this.ssh('sudo', 'mkdir', '-p', credfwdDir);
-      await this.writeFile(credfwdFile, fileContents, 0o644);
-      await this.writeFile('/usr/local/bin/docker-credential-rancher-desktop', DOCKER_CREDENTIAL_SCRIPT, 0o755);
-      await this.ssh('sudo', 'mkdir', '/root/.docker');
-      await this.writeFile('/root/.docker/config.json', configContents, 0o644);
+      await this.ssh('sudo', 'mkdir', '-p', CREDFWD_DIR);
+      await this.writeFile(CREDFWD_FILE, fileContents, 0o644);
+      await this.writeFile(DOCKER_CREDENTIAL_PATH, DOCKER_CREDENTIAL_SCRIPT, 0o755);
+      try {
+        const existingConfig = JSON.parse(await this.limaWithCapture('shell', '0', 'sudo', 'cat', ROOT_DOCKER_CONFIG_PATH));
```

Comment:
Lima likes to set working directory, which might not exist in the VM, right?

Reply:
Yes, it does, but you can overwrite with e.g. `--workdir=/`. I just took a look at `lima.ts` and see many instances that use `--workdir=.`, which is redundant, as it is already the default, and doesn't help with the problem that the local directory may not exist inside the VM:

```console
$ cd /Applications/Rancher\ Desktop.app

$ limactl shell alpine uname -a
bash: line 1: cd: /Applications/Rancher Desktop.app: No such file or directory
Linux lima-alpine 5.15.0-25-generic #25-Ubuntu SMP Wed Mar 30 15:54:22 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

Reply:
It isn't needed in this case, because it's calling `cat` with an absolute path

Reply:
No, `--workdir=.` actually works, because instead of doing `cd <something>` it does `cd .` which is a no-op. (This is _not_ the same as not having `--workdir`, which does `cd <cwd on the host>`.)

Either way, it _does_ need a `--workdir` of some sort, because the working directory on the host isn't guaranteed to exist in the VM.

Reply:
> No, --workdir=. actually works, because instead of doing cd <something> it does cd . which is a no-op.

Ah, right, it evaluates `.` inside the VM and not on the host. Thanks!

So yeah, the same thing should work here as well.

Reply:
> No, --workdir=. actually works, because instead of doing cd <something> it does cd . which is a no-op. 

It is needed because the current directory on the host may not exist inside the VM. See my example above:

```
bash: line 1: cd: /Applications/Rancher Desktop.app: No such file or directory
```

This is unrelated to whatever command you run; `limactl shell` tries to maintain the illusion that you are running commands on the host, so changes into the same directory inside the VM, which only works if the host directory has been mounted.

Reply:
Updated with `--workdir=.`